modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/SourceLookUpProvider.scala

```diff
@@ -13,6 +13,9 @@ private[debugadapter] final class SourceLookUpProvider(
     sourceUriToClassPathEntry: Map[URI, ClassEntryLookUp],
     fqcnToClassPathEntry: Map[String, ClassEntryLookUp]
 ) extends ISourceLookUpProvider {
+  private val classSearch = new SourceEntrySearchMap
+  for (entries <- classPathEntries; fqcn <- entries.fullyQualifiedNames) classSearch.insert(fqcn)
```

Comment:
I would prefer to use an immutable data structure:

```suggestion
val classSearch: Map[String, Seq[String]] = classPathEntries
  .flatMap(_.fullyQualifiedNames)
  .groupBy(getInnerType)
```